LdapInterop/UserSynchronizer.php

```diff
@@ -197,6 +197,43 @@ public function synchronizePiwikAccessFromLdap($piwikLogin, $ldapUser)
         }
     }
 
+    /**
+     * Purges a supplied Matomo login from DB based on LDAP entity absence.
+     *
+     * @param string $matomoLogin The username of the user who will be purged.
+     * @return string The Matomo user login that was purged.
+     */
+    public function purgeLdapUser($matomoLogin)
+    {
+        $userMapper = $this->userMapper;
+        $usersManagerAPI = $this->usersManagerApi;
+        $userModel = $this->userModel;
+        $logger = $this->logger;
+        return Access::doAsSuperUser(function () use ($matomoLogin, $userMapper, $usersManagerAPI, $userModel, $logger) {
```

Comment:
Can you remove this Access::doAsSuperUser() call? Ideally, we allow the caller to determine if they should force superuser access. Commands are already run w/ superuser privilege so there shouldn't be a need for this call.

Reply:
My code is heavily inspired from `synchronizeLdapUser`. Why other calls to this would be necessary and not this one?

Reply:
You're right, that should probably change too. If you feel up to it, feel free to change those methods as well.

Reply:
Same here: received an email notification for a comment which does not appear.
> You're right, that should probably change too. If you feel up to it, feel free to change those methods as well.